mission_control/src/behaviors/abort_behavior.cpp

```
@@ -0,0 +1,113 @@
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+// Original version: Christopher Scianna Christopher.Scianna@us.QinetiQ.com
+#include <string>
+
+#include "mission_control/behaviors/abort_behavior.h"
+
+using mission_control::AbortBehavior;
+
+AbortBehavior::AbortBehavior(const std::string& name,
+                                             const BT::NodeConfiguration& config)
+    : Behavior(name, config)
+{
+  subStateData_ =
+      nodeHandle_.subscribe("/state", 1, &AbortBehavior::stateDataCallback, this);
+
+  subThrusterRPM_ = nodeHandle_.subscribe("/thruster_control/report_rpm", 1,
+                                          &AbortBehavior::thrusterRPMCallback, this);
+
+  attitudeServoBehaviorPub_ =
+      nodeHandle_.advertise<mission_control::AttitudeServo>("/mngr/attitude_servo", 1);
+
+}
+
+BT::NodeStatus AbortBehavior::behaviorRunningProcess()
+{
+  if (!abortMsgHasBeenPublished_)
+  {
+    getInput<double>("roll", roll_);
+    getInput<double>("pitch", pitch_);
+    getInput<double>("yaw", yaw_);
+    getInput<double>("speed_knots", speedKnots_);
+    getInput<double>("roll_tol", rollTolerance_);
+    getInput<double>("pitch_tol", pitchTolerance_);
+    getInput<double>("yaw_tol", yawTolerance_);
```

@patapufate since we are adding a custom behavior to get the vehicle to reach the surface, do we need all these parameters? All are fixed, no?

---

Now are fixed, but we could use in the future the abort behavior in the tree structure, instead of being hard-coded.